pkgs/applications/audio/neothesia/default.nix

```
@@ -0,0 +1,93 @@
+{ lib
+, rustPlatform
+, fetchFromGitHub
+, cli ? false
+, pkg-config
+, cmake
+, glib
+, alsa-lib
+, atkmm
+, gtk3
+, wrapGAppsHook
+, ffmpeg-full
+, vulkan-loader
+, makeWrapper
+, fontconfig
+, noto-fonts
+, libclang
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "neothesia" + (if cli then "-cli" else "");
+  version = "unstable-2023-09-23";
+
+  src = fetchFromGitHub {
+    owner = "PolyMeilex";
+    repo = "Neothesia";
+    rev = "d724865059e876f14a3b92ba18a1c5dbe3485131";
+    hash = "sha256-CnVfv0iCuUgZcmtpxg8+wgG7r8HsLdKgjCbbNu5o76c=";
+  };
+
+  strictDeps = true;
+  nativeBuildInputs = [
+    pkg-config
+    cmake
+    atkmm
+    wrapGAppsHook
+  ] ++ (if cli then [
+    ffmpeg-full
+    rustPlatform.bindgenHook
+  ] else [
+    #fontconfig
+  ]);
+  buildInputs = [
+    gtk3
+    alsa-lib
+    atkmm
+  ] ++ (if cli then [
+    ffmpeg-full
+  ] else [
+    #fontconfig
+  ]);
+
+  cargoLock = {
+    lockFile = "${src}/Cargo.lock";
+    outputHashes = {
+      "mpeg_encoder-0.2.1" = "sha256-+BNZZ1FIr1374n8Zs1mww2w3eWHOH6ENOTYXz9RT2Ck=";
+    };
+  };
+
+  cargoBuildFlags = [ "--package" pname ];
+
+  #env.LIBCLANG_PATH = "${lib.makeLibraryPath [ libclang.lib ]}";
+
+  postInstall = (if cli then "" else ''
+    install -D flatpak/com.github.polymeilex.neothesia.desktop -t $out/share/applications
+    install -D flatpak/com.github.polymeilex.neothesia.png -t $out/share/icons/hicolor/256x256/apps
+  '');
+
+  preFixup = ''
+    gappsWrapperArgs+=(
+      --prefix LD_LIBRARY_PATH : ${lib.makeLibraryPath [ vulkan-loader ]}
+    )
+  '';
+
+  meta = with lib; {
+    description = (if cli then "A CLI program encoding visualisations of MIDI files into video" else "A GUI program visualising MIDI input and files by piano with sound");
+    longDescription = (if cli then ''
+      neothesia-cli is the CLI version of Neothesia.
+
+      When executing the program, you input the path to the MIDI file as the first argument. It will then render the notes flying towards a piano where a key lights up when a note is played. This is encoded into a video file.
+    '' else ''
+      Neothesia is a GUI program which can play MIDI files, visualising the notes on a piano it plays for you or which you play with MIDI input. Before the notes are played, they fly towars the piano.
+
+      It plays the notes through the sound output too. The default instrument sound can be changed by loading SoundFont files.
+
+      Neothesia may help you to learn how to play on the piano.
+    '');
+    homepage = "https://github.com/PolyMeilex/Neothesia";
+    license = licenses.gpl3;
```

```suggestion
    license = licenses.gpl3Only;
```

We prefer either `gpl3Plus` if the project is explicit later versions of GPL can apply or `gpl3Only` when they do not. This case seems to be `gpl3Only`